widget/embedded/src/utils/configs.ts

```diff
@@ -38,3 +43,63 @@ export const DEFAULT_SECONDARY_RADIUS = 25;
 export const DEFAULT_FONT_FAMILY = 'Roboto';
 
 export const THEME_CLASS_NAME_PREFIX = `theme-widget`;
+
+export const isExcluded = (
+  tokens: { [blockchain: string]: Tokens },
+  blockchain: string
+) => tokens[blockchain].isExcluded;
+
+export const handleHasToken = (
+  token: Token,
+  tokensConfig:
+    | {
+        [blockchain: string]: Tokens;
+      }
+    | Asset[]
+) => {
+  let hasToken = false;
+  if (Array.isArray(tokensConfig)) {
+    hasToken = !tokensConfig.some((asset) => tokensAreEqual(asset, token));
+  } else {
+    if (tokensConfig[token.blockchain]) {
+      hasToken = tokensConfig[token.blockchain].tokens.some((asset) =>
+        tokensAreEqual(asset, token)
+      );
+    }
+  }
+
+  return hasToken;
+};
+
+export const shouldChangeSelectedToken = (
+  token: Token | null,
+  tokensConfig?:
+    | {
+        [blockchain: string]: Tokens;
+      }
+    | Asset[]
+) => {
+  let hasToken = false;
+  if (tokensConfig && token) {
+    hasToken = handleHasToken(token, tokensConfig);
+    if (!Array.isArray(tokensConfig) && tokensConfig[token.blockchain]) {
+      if (
+        (!isExcluded(tokensConfig, token.blockchain) && !hasToken) ||
+        (isExcluded(tokensConfig, token.blockchain) && hasToken)
+      ) {
+        return true;
+      }
+    }
+  }
+  return hasToken;
+};
```

Comment:
I think this logic is duplicate of what we have in lines  +94 to +112 of data store.

Reply:
fixed